backend/selectgen.ml

```diff
@@ -1335,55 +1240,44 @@ method emit_expr_aux (env:environment) exp ~bound_name :
                     (env_handler.trap_stack,
                      instr_cons_debug (Iop Imove) [|Proc.loc_exn_bucket|] rv
                        dbg
-                       (end_region s2#extract))))
+                       s2#extract)))
           [||] [||];
         r
       in
       let env = env_add v rv env in
       begin match kind with
-      | Regular -> with_handler env e2
       | Delayed lbl ->
         begin match env_find_static_exception lbl env_body with
         | { traps_ref = { contents = Reachable ts; }; _} ->
           with_handler (env_set_trap_stack env ts) e2
         | { traps_ref = { contents = Unreachable; }; _ } ->
-          let unreachable =
+          let dummy_constant =
+            match r1 with
+            | None -> Ctuple []
```

Comment:
This line causes the error in the cmm tests.
The body doesn't return, but it might be joined with a branch that does, and we can't guess the type.
We need to find a simple expression that is non-returning; I'll try to come up with something.